File: 2017-2018/Lato/PO/List 05/Ex 2/src/Constant.java
import java.util.HashMap;

public class Constant extends Expression{
    double value;

    public Constant(double value){
        this.value = value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }

    public double Eval(HashMap<String, Double> vals) {
        return value;
    }
}

File: Term II/C++/Lista 6/Constant.cpp
#include "Constant.h"



Constant::Constant(std::string symbol, double value) : value(value)
{
	this->symbol = symbol;
}

Constant::~Constant()
{
}

double Constant::eval()
{
	return value;
}

std::string Constant::toString()
{
	return symbol;
}
